### DISCUSSION TITLE
Reuse HttpClient

### DISCUSSION THREAD

**Discussion by czullu** [pnp/pnpcore#971] [General]
Looking at PnPContext and PnPContextFactory implementation, i was wondering if there is/was a specific reason to not create httpClient in the PnPContextFactory and pass it to PnPContext in order to reuse the HttpClient.

The change i'm thinking of is, that we try to get HttpClient from dependency injection in PnPContextFactory and only create a new HttpClient if we can not get one. Then we pass the HttpClient on to PnPContext. In that way we can make sure that connection pooling is working.

Any thoughts against that change?

@jansenbe

**Comment by jansenbe** [comment]
Hi @czullu, 

The HttpClients are added as part of the PnPContextFactory, see https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Services/Core/PnPContextFactoryCollectionExtensions.cs#L72-L120. You can also get new `HttpClient` instance from the connection pool via dependency injection:

![image](https://user-images.githubusercontent.com/7451219/192702062-a852aae7-70dc-4275-8772-cb5653d76f04.png)

**Comment by czullu** [comment]
Hi Bert
Thank you for response – the line which confuse me is in PnPContext as I think it’s not respecting the injection.

***@***.***

With best regards
Christian
